domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/PerformanceMetricsLogger.kt

```diff
@@ -0,0 +1,260 @@
+package org.oppia.android.domain.oppialogger.analytics
+
+import org.oppia.android.app.model.OppiaMetricLog
+import org.oppia.android.domain.oppialogger.ApplicationStartupListener
+import org.oppia.android.util.logging.performancemetrics.PerformanceMetricsUtils
+import org.oppia.android.util.system.OppiaClock
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class PerformanceMetricsLogger @Inject constructor(
+  private val performanceMetricsController: PerformanceMetricsController,
+  private val performanceMetricsUtils: PerformanceMetricsUtils,
+  private val oppiaClock: OppiaClock
+) : ApplicationStartupListener {
+
+  /**
+   * Use a large Long value such that the time difference based on any timestamp will be negative
+   * and thus ignored until the app records initial time during [onCreate].
+   */
+  private var firstTimestamp: Long = Long.MAX_VALUE
+
+  override fun onCreate() {
+    firstTimestamp = oppiaClock.getCurrentTimeMs()
+  }
+
+  /** Logs the apk size of the application as a performance metric. */
+  fun logApkSize() {
+    performanceMetricsController.logLowPriorityMetricEvent(
+      oppiaClock.getCurrentTimeMs(),
+      OppiaMetricLog.CurrentScreen.HOME_SCREEN,
+      createApkSizeLoggableMetric(performanceMetricsUtils.getApkSize())
+    )
+  }
+
+  /** Logs the storage usage of the application as a performance metric. */
+  fun logStorageUsage() {
+    performanceMetricsController.logLowPriorityMetricEvent(
+      oppiaClock.getCurrentTimeMs(),
+      OppiaMetricLog.CurrentScreen.HOME_SCREEN,
+      createStorageUsageLoggableMetric(performanceMetricsUtils.getUsedStorage())
+    )
+  }
+
+  /** Logs the startup latency of the application as a performance metric. */
```

Comment:
Per https://developer.android.com/topic/performance/vitals/launch-time could we rely on an application-wide activity observer that saves the time the first onStart is called and use that as opposed to application's own start method being called in order to compute the cold start latency?

Reply:
The application-wide activity observer will also record the time at the onCreate() method call of the application class and we're also doing the same here -- in a simpler way. Is there a particular downside to what is being done here?

Reply:
Doesn't this require making sure that we call ``logStartupLatency`` at precisely the correct time? If we change our app entrypoints in the future, or change the behavior of classes like SplashActivity, this method may not work exactly as expected. It seems more robust to always log at the _first_ activity's onCreate regardless of what activity it is.

Reply:
okay, here's what I'm thinking. There are two points at which we've to log timestamps for calculating startup latency: entry point and first activity's onResume(). For the entry point, the first method called at application startup is OppiaApplication's onCreate() so that becomes our entry point. And in case of the first activity, we can register activityLifecycleCallbacks to track our activity's onResume(). 

Alternatively, we can also use the LifeCycle Observer to observe onCreate() lifecycle event of the application (like we do in case of foregrounding and backgrounding) and then log the first timestamp -- this is however, same as Application Class' onCreate(). 

Does this approach look good to you? I think the current approach is on the right path and you'll see how we're depending on the first activity's onResume() instead of SplashActivity's onResume() in the logging PR.

Reply:
Yes I think this sounds good (using ActivityLifecycleCallbacks).